README.md

```diff
@@ -1 +1,216 @@
-# pbxd
\ No newline at end of file
+# pbxd
+
+pbxd is a web service that provides an API to an Avaya PBX.
+
+It logs in to the SAT terminal of the PBX, then runs commands on the PBX and
+returns the results to the client.
+
+The number of gunicorn workers determines how many simultaneus logins are made
+to the PBX system.
+
+
+## Configuration
+
+Environment variables:
+
+    PBX_NAME=uw01
+    PBX_COMMAND_TIMEOUT=300
+    PBXD_CONF=pbxd_conf/pbxd_uw01_conf.json
+
+Secrets are loaded from a JSON config file like this:
+
+    {
+        "connection_command": "/usr/bin/ssh -o 'StrictHostKeyChecking no' -o 'ServerAliveInterval -p 5022 -l username
+        ip_address_of_pbx",
+        "pbx_username": "username",
+        "pbx_password": "password"
+    }
+
+## Access control
+
+Restricting access to authorized users must be done by a proxy server like Nginx. Typically this will require X.509 certificates or host IP addresses.
+
+
+## Dependencies
+
+There are some python packages required for pbxd in the setup.py file and
+it is good to keep them updated. pip-tools is used here to update the
+requirements.txt file and help ensure predictable builds.
+
+    deactivate; rm -rf venv
+    python3 -m venv venv && . venv/bin/activate
+    pip install pip-tools
+    pip-compile
+
+
+## Running
+
+### Run in a virtual environment
+
+    python3 -m venv venv && . venv/bin/activate
+    pip install -r requirements.txt
+    pip install -e .
+
+    PBX_NAME=uw01 \
+    PBX_COMMAND_TIMEOUT=300 \
+    PBXD_CONF=pbxd_conf/pbxd_uw01_conf.json \
+    gunicorn "pbxd.app:load()" -b localhost:8000 --access-logfile - --log-level INFO --reload
+
+### Run in Docker container
+
+    docker-compose build
+    docker-compose up --abort-on-container-exit
+
+
+
+## Client API
+
+Clients use pbxd to run a command on the PBX and receive the results.
+
+The PBX need to know what terminal type to use:
+- `ossi` is the terminal type that returns fields with unique IDs. You can find the
+IDs by doing a display command and use them in a change command.
+- `vt220` is the terminal type that returns screen snapshots. You only use this
+for display commands.
```

Comment:
What clients require the `vt220` support?

Reply:
vt220 is the screen view that users experience when they log in to the PBX. They might actually still use 4410 but it looks almost the same as VT220. We have a few clients that show PBX screens to the users.

Reply:
Sure. Which clients are those? I'm not really familiar with the pbx-supporting thingies we own.

Reply:
I already forgot this is open source, maybe reply in slack.